lib/auth/userloginstate/generator.go

```diff
@@ -120,6 +137,11 @@ func (g *Generator) Generate(ctx context.Context, user types.User) (*userloginst
 		return nil, trace.Wrap(err)
 	}
 
+	// Emit the usage event metadata.
+	if err := g.emitUsageEvent(ctx, user, uls); err != nil {
+		g.log.WithError(err).Debug("Error emitting usage event during user login state generation")
```

Comment:
`WithError` will dump a stack trace which implies that you want to draw attention to this in the logs, but debug level implies that it's not critical. I would either remove the `WithError` or make the level warn.